src/GitVersion.Core.Tests/Configuration/ConfigProviderTests.CanWriteOutEffectiveConfiguration.approved.txt

```diff
@@ -57,10 +57,8 @@ branches:
     pre-release-weight: 30000
   feature:
     mode: ContinuousDelivery
-    tag: useBranchName
+    tag: '{BranchName}'
```

Comment:
Yep that's true. Here are some aspects I need to explain. Please let me know if you are not agree with my argumentation or if you have any better solutions:

1. I have removed the default initialization behavior of BranchConfig and the usage of BranchConfig.CreateDefaultBranchConfig. This was necessary because otherwise we cannot inherit the properties Tag, PreventIncrementOfMergedBranchVersion, TrackMergeTarget, TracksReleaseBranches, IsReleaseBranch, IsMainline when specifying the IncrementStrategy to Increment.Inherit. You see that the changes only effects this branch configuration where the IncrementStrategy flag is set to inherit.
2. The tag useBranchName name is equivalent to {BranchName}. In my opinion useBranchName is a magic string and not so intuitive. Better using templates like {BranchName} which is clear that this will be replaced with the name of the branch. I can think of that we have other properties in future.
3. Hotfix branches can be created from release (new), main, support, hotfix (new) but not from develop.
4. The default behavior if you not defining IncrementStrategy is Inherit

Reply:
Ok, sounds good.